docs/zh_CN/guide/ai/tensorflow_lite_external_function_tutorial.md

```diff
@@ -0,0 +1,135 @@
+# 使用外部函数运行 AI 算法
+
+[LF Edge eKuiper](https://www.lfedge.org/projects/ekuiper/) 是一款边缘轻量级物联网数据分析/流软件，可在各种资源受限的物联网设备上运行。
+
+[TensorFlow Lite](https://www.tensorflow.org/lite/guide) 是一组帮助开发人员在移动端、嵌入式和物联网设备上运行 TensorFlow
+模型的工具，它使得设备上的机器学习预测具有低延迟和较小的二进制容量。
+
+通过集成 eKuiper 和 TensorFlow Lite，用户可以使用预建的 TensorFlow 模型通过 AI 分析流中的数据。
+在本教程中，我们将通过预先训练的图像识别 TensorFlow 模型, 来带您构建一个 eKuiper 外部函数插件来标记边缘生成的图片。
+通过使用外部函数，eKuiper和外部函数可以运行在完全不同的进程或主机中，这意味着 eKuiper 和外部函数可以有不同的生命周期，更重要的是，外部函数
+可以为除 eKuiper 以外的其他系统程序提供服务。
+
+
+## 先决条件
+
+外部功能插件将是一个 GRPC 服务器，因此用户应该了解 GRPC。本教程将给出设置 GRPC 服务器的示例代码。
+用户可以在[这里](https://github.com/lf-edge/ekuiper/blob/master/docs/resources/pythonGRPC.zip)下载示例代码。
+
+用户还需要具备 Docker 的基本知识。
+
+## 开发外部函数
+
+在示例代码中，GRPC Server提供了 ``label`` 方法，用户只需要编写一个接口描述文件，注册到 eKuiper 中即可。然后eKuiper就可以调用 GRPC 方法,
+就像内置函数一样。 ``label`` 方法由 ``tflite_runtime`` 图像分类提供支持，有关更多详细信息，请查看示例代码中的 `label.py` 文件。
+
+这是提供服务的外部功能的 proto 文件描述。 ``label`` 方法的参数应该是 base64 编码的图像。
+
+```proto
+syntax = "proto3";
+
+package sample;
+
+// The algorithms service definition.
+service Algorithms {
+  rpc label(LabelRequest) returns(LabelReply) {}
+}
+
+// The request message containing the base64 encoded image.
+message LabelRequest {
+  string base64_img = 1;
+}
+
+message LabelResult {
+  float  confidence = 1;
+  string label = 2;
+}
+
+// The response message containing the greetings
+message LabelReply {
+  repeated LabelResult results = 1;
+}
+```
+
+## 构建和运行 GRPC Server
+
+我们提供Dockerfile来构建GRPC服务器，进入示例代码的根路径，运行以下命令来构建GRPC docker镜像
+
+```shell
+ docker build  -t test:1.1.1 -f deploy/Dockerfile-slim-python .
+```
+
+然后通过以下命令来启动服务
+
+```shell
+ docker run -d  -p 50051:50051 --name rpc-test test:1.1.1
+```
+
+现在，GRPC 服务器在 50051 端口上提供服务。
+
+## 打包并注册外部函数
+
+### 打包
+
+将 GRPC server 中服务的 json 描述文件和 proto 文件打包成zip。 zip 文件中的文件结构应如下所示：
```

Comment:
Please check the case of English words, and make sure spaces between English and Chinese characters.